src/js/view/utils/resize-listener.js

```diff
@@ -49,26 +54,29 @@ export default class ResizeListener {
         this.element = element;
         this.callback = callback;
         this.resizeRaf = -1;
-        this.lastWidth = 0;
-        this.currentWidth = element.offsetWidth;
+        this.lastWidth = this.currentWidth = 0;
         this.scrollListener = (e) => {
-            let resizeRaf = this.resizeRaf;
-            if (resizeRaf) {
-                cancelAnimationFrame(resizeRaf);
-            }
-            resizeRaf = requestAnimationFrame(() => {
+            cancelAnimationFrame(this.resizeRaf);
+            this.resizeRaf = requestAnimationFrame(() => {
                 const currentWidth = this.currentWidth = element.offsetWidth;
                 if (this.lastWidth === currentWidth) {
                     return;
                 }
                 this.callback(e, currentWidth);
             });
-            this.resizeRaf = resizeRaf;
             this.resetTriggers();
         };
 
-        element.addEventListener('scroll', this.scrollListener, true);
-        this.resetTriggers();
+        requestAnimationFrame(() => {
+            // TODO: this should be moved to the widget. The player element is always relative.
```

Comment:
assume you mean shelf widget here @robwalch ?

Reply:
Yep. The player must be styled `position: relative` or the absolute positioned elements contained within will fill the page: https://github.com/jwplayer/jwplayer/blob/master/src/css/jwplayer/imports/jwplayerlayout.less#L6